starter-code/src/viking.js

```diff
@@ -1,11 +1,104 @@
 // Soldier
-function Soldier() {}
+function Soldier(health, strength) {
+    this.health = health;
+    this.strength = strength;
+}
+
+Soldier.prototype.attack=function(){
+    return this.strength;
+}
+
+Soldier.prototype.receiveDamage=function(damage){
+    this.health -= damage;
+}
 
 // Viking
-function Viking() {}
+Viking.prototype=Object.create(Soldier.prototype);
+
+function Viking (name,health,strength){
+    this.name=name;
+    this.health=health;
+    this.strength=strength;
+}
+
+Viking.prototype.attack=function(){
+    return this.strength;
+}
+
+Viking.prototype.receiveDamage=function(damage){
+    this.health -= damage;
+    if (this.health > 0){
+        return this.name + " has received " + damage + " points of damage";
+    }
+    if (this.health <= 0){
+        return this.name + " has died in act of combat";
+    }
+}
+
+Viking.prototype.battleCry = function(){
+    return "Odin Owns You All!"
+}
 
 // Saxon
-function Saxon() {}
+Saxon.prototype=Object.create(Soldier.prototype);
+
+function Saxon (health,strength){
+    this.health=health;
+    this.strength=strength;
+}
+
```

Comment:
Same remark than for the Viking, the Saxon constructor should have looked like this:
```javascript
function Saxon( health, strength ) {
    Soldier.call( this, health, strength );
};
```